interface/repository/pokemon_repository.go

```diff
@@ -0,0 +1,102 @@
+package repository
+
+import (
+	"encoding/csv"
+	"errors"
+	"os"
+	"strconv"
+
+	"github.com/alexis-wizeline/ondemand-go-bootcamp/domain/model"
+	repo "github.com/alexis-wizeline/ondemand-go-bootcamp/usecase/repository"
+)
+
+type pokemonRepository struct {
+}
+
+func NewPokemonRepository() repo.PokemonRepository {
+	return pokemonRepository{}
+}
+
+func (p pokemonRepository) GetPokemons() ([]*model.Pokemon, error) {
+	pokemons, err := getAllPokemons()
+
+	if err != nil {
+		return pokemons, err
+	}
+
+	return pokemons, nil
+}
+
+func (p pokemonRepository) GetPokemonById(id uint64) (*model.Pokemon, error) {
+	pokemons, err := getAllPokemons()
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, pokemon := range pokemons {
+		if id == pokemon.ID {
+			return pokemon, nil
+		}
+	}
+
+	return nil, nil
+}
+
+func getAllPokemons() ([]*model.Pokemon, error) {
+	var pokemons []*model.Pokemon
+	rows, err := openAndGetCSVData()
+
+	if err != nil {
+		return pokemons, err
+	}
+
+	pokemons, err = transformRowsToPokemons(pokemons, rows)
+
+	if err != nil {
+		return pokemons, err
+	}
+
+	return pokemons, nil
+
+}
+
+func transformRowsToPokemons(pokemons []*model.Pokemon, rows [][]string) ([]*model.Pokemon, error) {
```

Comment:
you don't need to receive `pokemons []*model.Pokemon`. you can just declare and return the pokemons list in this function